src/openrct2-ui/windows/LoadSave.cpp

```diff
@@ -269,9 +269,18 @@ static void Select(const char* path)
     {
         case (LOADSAVETYPE_LOAD | LOADSAVETYPE_GAME):
             SetAndSaveConfigPath(gConfigGeneral.LastSaveGameDirectory, pathBuffer);
-            InvokeCallback(MODAL_RESULT_OK, pathBuffer);
-            window_close_by_class(WindowClass::Loadsave);
-            gfx_invalidate_screen();
+            if (Editor::LoadLandscape(pathBuffer))
```

Comment:
I think the `LoadLandscape` is not the best choice, as it only accounts for extensions compatible with landscape and doesn't account for `.sea` or `.sv7 (though I don't know if it should, should it @Gymnasiast?). `Context::LoadParkFromFile` seems more appropriate, though one might wonder if this is exactly the same thing that the callback is doing.

Reply:
I tested with Context:LoadParkFromFile originally, and it worked as well, but InvokeCallback also calls that indirectly as well, so I’m sure if that would be optimum.

Reply:
Yeah, that's what I thought, if it would be wasteful. @Gymnasiast can you weigh in on my first comment, please?

Reply:
I agree with you. Editor::LoadLandscape is made specifically to, well, load landscapes. In general, our load/save functions seem to have overlap. By the way, the point about loading twice (once here, once in the callback) appears to apply to all cases, whether you call Editor::LoadParkFromFile() or Context::LoadParkFromFile() or even another function.

Reply:
Yeah, there's also that, other places are already doing it twice.

Can you please change it accordingly then Henry?

Reply:
Changed the check from LoadLandscape to LoadParkFromFile